src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -373,6 +374,11 @@ def column_states(self, value: ColumnStateDict):
         """
         self._column_states = deepcopy(value)
 
+    @abstractmethod
+    def referred_ctes(self) -> Set[str]:
+        """Return the set of ctes referred by the current selectable node"""
```

Comment:
is this the set of CTE's that are referenced inside the subtree rooted at this node? can we update the description to include we mean subtree here.

Reply:
yes, added